src/Views/VPNPage.vala

```diff
@@ -80,29 +62,28 @@ public class Network.VPNPage : Network.Widgets.Page {
         add_button.tooltip_text = _("Add VPN Connection…");
         add_button.clicked.connect (() => {
             add_button.sensitive = false;
+            add_button.sensitive = create_connection ();
+        });
 
-            try {
-                var command = AppInfo.create_from_commandline ("nm-connection-editor --create --type=vpn", null, GLib.AppInfoCreateFlags.NONE);
-                command.launch (null, null);
+        var remove_button = new Gtk.ToolButton (new Gtk.Image.from_icon_name ("list-remove-symbolic", Gtk.IconSize.SMALL_TOOLBAR), null);
+        remove_button.tooltip_text = _("Forget selected VPN…");
+        remove_button.sensitive = false;
+        remove_button.clicked.connect (remove_button_cb);
 
-                add_button.sensitive = true;
-            } catch (Error e) {
-                var dialog = new Granite.MessageDialog (
-                    _("Failed To Run Connection Editor"),
-                    _("The program \"nm-connection-editor\" may not be installed."),
-                    new ThemedIcon ("dialog-error"),
-                    Gtk.ButtonsType.CLOSE
-                );
-                dialog.show_error_details (e.message);
-                dialog.transient_for = (Gtk.Window) get_toplevel ();
-                dialog.run ();
-                dialog.destroy ();
-            }
+        remove_vpn_toast.default_action.connect (() => {
+            GLib.Source.remove (timeout_id);
+            timeout_id = 0;
+            sel_row.show ();
         });
 
-        toolbar.add (add_button);
+        var edit_connections_button = new Gtk.ToolButton (new Gtk.Image.from_icon_name ("preferences-system-symbolic", Gtk.IconSize.SMALL_TOOLBAR), null);
+        edit_connections_button.tooltip_text = _("Edit VPN connections…");
+        edit_connections_button.sensitive = false;
```

Comment:
It looks like we can still run the connection editor even if the selected row is null, and we already handle that case. So we probably don't need to ever make it insensitive

Reply:
see https://github.com/elementary/switchboard-plug-network/pull/210